### PR TITLE
refactor: update `blas/ext/base/dssumpw` to follow current project conventions

### LINKED ISSUE #1503: [RFC]: refactor `blas/ext/base/dssumpw` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dssumpw`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dssumpw) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by soumajit23**
Resolves # 1503

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dssumpw to follow current project conventions.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves https://github.com/stdlib-js/stdlib/issues/1503

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/addon.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, alpha, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, Y, N, strideY, argv, 4 );
+	c_saxpy( N, alpha, X, strideX, Y, strideY );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/addon.c]: This implementation is incorrect. See the `addon.cpp` file and update accordingly.

Comment (soumajit23) [reply lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/addon.c]: Updated `addon.c` file as instructed. Please review. Thanks!

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2020 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md]:
```suggestion
Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/dssumpw.c [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/dssumpw.c:4]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/dssumpw.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.dssumpw.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.dssumpw.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.ndarray.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.ndarray.native.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.dssumpw.native.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/test/test.dssumpw.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/Makefile [hunk 9]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/addon.c [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, (double)stdlib_strided_dssumpw( N, X, stride ), &v );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/src/addon.c]: Cast not needed:

```suggestion
	napi_status status = napi_create_double( env, stdlib_strided_dssumpw( N, X, stride ), &v );
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/ndarray.native.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/native.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/main.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/main.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/include.gypi [hunk 14]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/include.gypi:3]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/include/stdlib/blas/ext/base/dssumpw.h [hunk 15]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/include/stdlib/blas/ext/base/dssumpw.h]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/dssumpw.js [hunk 16]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/dssumpw.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/dssumpw.native.js [hunk 17]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/dssumpw.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/index.js [hunk 18]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/lib/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/include.gypi (hunk 14 above)
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/include.gypi]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/index.js [hunk 19]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/c/example.c [hunk 20]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/c/example.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/c/Makefile [hunk 21]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/c/Makefile]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/docs/types/test.ts [hunk 22]
```diff
@@ -1,7 +1,7 @@
 /*
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/docs/types/test.ts]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/docs/types/index.d.ts [hunk 23]
```diff
@@ -1,7 +1,7 @@
 /*
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/docs/types/index.d.ts]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/c/Makefile [hunk 24]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/c/Makefile]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/c/benchmark.length.c [hunk 25]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/c/benchmark.length.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/binding.gyp [hunk 26]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/binding.gyp]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.ndarray.native.js [hunk 27]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.ndarray.js [hunk 28]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.native.js [hunk 29]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.js [hunk 30]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/c/Makefile [hunk 31]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2020 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/benchmark/c/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/binding.gyp [hunk 32]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2020 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/binding.gyp]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/c/Makefile [hunk 33]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2020 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/examples/c/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

**Comment by soumajit23** [comment]
@Planeshifter are there any more changes that are required?
If not, please merge this PR since I plan on applying for GSoC, and a merged PR is required to meet the patch requirements.
Thanks!

**Comment by kgryte** [comment]
@soumajit23 Looks like CI is failing for this PR. Until test and benchmark failures are addressed, this PR won't be able to move forward.

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**